Description: fix for Debian bug #741451
 tiffcp crashes when converting JPEG-encoded TIFF to a different
 encoding (like none or lzw). For example this will probably fail:
 .
        tiffcp -c none jpeg_encoded_file.tif output.tif
 .
 The reason is that when the input file contains JPEG data,
 the tiffcp code forces conversion to RGB space. However,
 the output normally inherits YCbCr subsampling parameters
 from the input, which leads to a smaller working buffer
 than necessary. The buffer is subsequently overrun inside
 cpStripToTile() (called from writeBufferToContigTiles).
 Note that the resulting TIFF file would be scrambled even
 if tiffcp wouldn't crash, since the output file would contain
 RGB data intepreted as subsampled YCbCr values.
 .
 This patch fixes the problem by forcing RGB space on the output
 TIF if the input is JPEG-encoded and output is *not* JPEG-encoded.
Origin: upstream, http://bugzilla.maptools.org/show_bug.cgi?id=2480
Author: Tomasz Buchert <tomasz.buchert@inria.fr>
Bug: http://bugzilla.maptools.org/show_bug.cgi?id=2480
Bug-Debian: http://bugs.debian.org/741451

--- a/tools/tiffcp.c
+++ b/tools/tiffcp.c
@@ -629,6 +629,12 @@
 		TIFFSetField(out, TIFFTAG_PHOTOMETRIC,
 		    samplesperpixel == 1 ?
 		    PHOTOMETRIC_LOGL : PHOTOMETRIC_LOGLUV);
+	else if (input_compression == COMPRESSION_JPEG &&
+		samplesperpixel == 3) {
+		/* RGB conversion was forced above
+		   hence the output will be of the same type */
+		TIFFSetField(out, TIFFTAG_PHOTOMETRIC, PHOTOMETRIC_RGB);
+	}
 	else
 		CopyTag(TIFFTAG_PHOTOMETRIC, 1, TIFF_SHORT);
 	if (fillorder != 0)
